engine/runtime-compiler/src/test/java/org/enso/compiler/test/mock/WithMockCompilerContext.java

```diff
@@ -0,0 +1,160 @@
+    return new Builder().build();
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public MockPackageRepository getRepo() {
+    return repo;
+  }
+
+  public Compiler getCompiler() {
+    return compiler;
+  }
+
+  public Package<FileObject> createPackage(LibraryName pkgName, SourceModule module) {
+    return repo.createPackage(pkgName, module);
+  }
+
+  /**
+   * Create a package with the given set of source modules in the virtual file system.
+   *
+   * @param pkgName
+   * @param modules
+   * @return Created package, or null if an error occurred.
+   */
+  public Package<FileObject> createPackage(LibraryName pkgName, Set<SourceModule> modules) {
+    return repo.createPackage(pkgName, modules);
+  }
+
+  public void registerMainProjectPackage(LibraryName pkgName, Package<FileObject> mainPkg) {
+    repo.registerMainProjectPackage(pkgName, MockPackageRepository.castVirtualPkg(mainPkg));
+  }
+
+  public List<Module> getLoadedModules() {
+    return CollectionConverters.asJava(repo.getLoadedModules());
+  }
+
+  @Override
+  public Statement apply(Statement base, Description description) {
+    return new CustomStatement(base);
+  }
+
+  public final class CustomStatement extends Statement {
+    private final Statement base;
+
+    CustomStatement(Statement base) {
+      this.base = base;
+    }
+
+    @Override
+    public void evaluate() throws Throwable {
+      try {
+        base.evaluate();
+      } catch (Throwable e) {
+        printCompilerOutput();
+        printAllVfsFiles();
+        throw e;
+      } finally {
+        repo.getVfs().deleteAll();
+      }
+    }
+
+    private void printCompilerOutput() {
+      System.err.println("=== Compiler Output ===");
+      System.err.println(out);
+      System.err.println("=== End of Compiler Output ===");
+    }
+
+    private void printAllVfsFiles() {
+      var allFiles = repo.listAllFilesInVfs();
+      System.err.println("=== All files in VFS === ");
+      System.err.println(allFiles);
+      System.err.println("=== End of VFS === ");
+    }
+  }
+
+  public static final class Builder {
+    private boolean enableWarnings = true;
+    private boolean enableStaticTypeInference = false;
+    private boolean isStringErrors = true;
+    private boolean enableLinting = true;
+
+    Builder() {}
+
+    public Builder enableWarnings(boolean enable) {
+      this.enableWarnings = enable;
+      return this;
+    }
+
+    public Builder enableStaticTypeInference(boolean enable) {
+      this.enableStaticTypeInference = enable;
+      return this;
+    }
+
+    public Builder isStringErrors(boolean enable) {
+      this.isStringErrors = enable;
+      return this;
+    }
+
+    public Builder enableLinting(boolean enable) {
+      this.enableLinting = enable;
+      return this;
+    }
+
+    public WithMockCompilerContext build() {
+      var out = new ByteArrayOutputStream();
+      var compilerCfg =
+          new CompilerConfig(
```

Comment:
I was also looking at `CompilerConfig`. It's use from Java isn't easy. You are mitigating that problem by hiding the config into new builder in tests.

Why not rewrite `CompilerConfig` into Java and turn it into builder pattern itself? Then you'd have:
```
public static WithMockCompilerContext create(CompilerConfig) {
```

Reply:
`CompilerConfig` refactored to Java in 341ca62ec2ef61d8aebeaed18f69b436351d4d90